### PR TITLE
Structure_Engine: Checking if property is null in AnalyticalBars

### LINKED ISSUE #1406: Structure_Engine: AnalyticalBars crash badly
#### Description:

Having plugged some `IFramingElements` into `AnalyticalBars` component, the latter crashed with multiple errors.

#### Test file(s):

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FBHoM%5FEngine%2FStructure%5FEngine%2FStructure%5FEngine%2DIssue1406%2DAnalyticalBarsCrash).

### PR DESCRIPTION
**Pull request by albinber**
### Issues addressed by this PR

Closes #1406

Checking if the property is null and displaying a more informative error message.

### CODE REVIEW

File: Structure_Engine/Query/AnalyticalBars.cs [hunk 1]
```diff
@@ -101,7 +101,12 @@ private static List<Bar> AnalyticalBars(BHP.FramingProperties.ConstantFramingPro
 
             ISectionProperty section;
 
-            if (convertedProps.ContainsKey(property))
+            if (property == null)
+            {
+                Engine.Base.Compute.RecordError("One or more properties are null. Please make sure all of the inputs have a value.");
+                return new List<Bar>();
+            }
+            else if (convertedProps.ContainsKey(property))
```
Comment (Tom-Kingstone) [inline Structure_Engine/Query/AnalyticalBars.cs:109]: Is this supposed to not return anything if a property is not given? If so I am fine to approve here. (as far as I can tell, after this, the method seems to expect a section derived from this property, Just asking for confirmation really)

File: Structure_Engine/Query/AnalyticalBars.cs [hunk 2]
```diff
@@ -101,7 +101,12 @@ private static List<Bar> AnalyticalBars(BHP.FramingProperties.ConstantFramingPro
 
             ISectionProperty section;
 
-            if (convertedProps.ContainsKey(property))
+            if (property == null)
+            {
+                Engine.Base.Compute.RecordError("One or more properties are null. Please make sure all of the inputs have a value.");
+                return new List<Bar>();
+            }
```
Comment (FraserGreenroyd) [inline Structure_Engine/Query/AnalyticalBars.cs:108]: I would put this as its own if statement after line 94 @albinber - as further in the method has shown (from your discussion with me on teams) that the `property` should not be null at any point really, which makes sense. So checking if it is null at the start of the method will avoid any crashes later one.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by Tom-Kingstone — APPROVED** [review]
After changes from discussion above, I am happy to approve.

**Comment by Tom-Kingstone** [comment]
@BHoMBot check core
@BHoMBot check compliance

File: Structure_Engine/Query/AnalyticalBars.cs [hunk 3]
```diff
@@ -93,6 +93,12 @@ private static List<Bar> AnalyticalBars(BHP.FramingProperties.ConstantFramingPro
                 return new List<Bar>();
             }
 
+            if (property == null)
+            {
+                Engine.Base.Compute.RecordError("One or more properties are null. Please make sure all of the inputs have a value.");
```
Comment (FraserGreenroyd) [inline Structure_Engine/Query/AnalyticalBars.cs:98]:
```suggestion
                Engine.Base.Compute.RecordError("The Constant Framing Property is null and as such, analytical bars cannot be constructed. Please ensure all of the inputs have a valid value to calculate the analytical bars.");
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — APPROVED** [review]
I have tested this with the script I was using yesterday during the historic issue review and this provides a much clearer error message to the user now as opposed to the previous error message which was generated by the `null` `property` object.

Thanks for picking up so quick @albinber 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: unit-tests

**Pull request merged**